From 3d31ebb9fee2973f96cdfafa51040d19e1f5d9b1 Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Tue, 30 Apr 2013 22:20:00 -0700
Subject: [PATCH 09/13] ARM: Ambarella: Add Hero3 Black LED devices

Add the platform devices for supporting the GPIO-based
LEDs on the Heror3 Black camera.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/mach-ambarella/init-filbert.c |   37 ++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm/mach-ambarella/init-filbert.c b/arch/arm/mach-ambarella/init-filbert.c
index ad7b727..eb6053a 100644
--- a/arch/arm/mach-ambarella/init-filbert.c
+++ b/arch/arm/mach-ambarella/init-filbert.c
@@ -43,6 +43,8 @@
 #include <linux/irq.h>
 #include <linux/interrupt.h>
 #include <linux/delay.h>
+#include <linux/gpio.h>
+#include <linux/leds.h>
 
 #include <plat/ambinput.h>
 
@@ -55,6 +57,40 @@ static struct platform_device ambarella_auc_codec0 = {
 };
 #endif
 
+static struct gpio_led hero3black_led_pins[] = {
+        {
+                .name                   = "front",
+                .default_trigger        = "heartbeat",
+                .gpio                   = 45,
+                .active_low             = 0,
+        },
+        {
+                .name                   = "top",
+                .default_trigger        = "off",
+                .gpio                   = 38,
+                .active_low             = 0,
+        },
+        {
+                .name                   = "back",
+                .default_trigger        = "off",
+                .gpio                   = 102,
+                .active_low             = 1,
+        },
+};
+
+static struct gpio_led_platform_data hero3black_led_data = {
+        .leds           = hero3black_led_pins,
+        .num_leds       = ARRAY_SIZE(hero3black_led_pins),
+};
+
+static struct platform_device hero3black_leds = {
+        .name   = "leds-gpio",
+        .id     = -1,
+        .dev    = {
+                .platform_data  = &hero3black_led_data,
+        }
+};
+
 /* ==========================================================================*/
 static struct platform_device *ambarella_devices[] __initdata = {
 	&ambarella_adc0,
@@ -83,6 +119,7 @@ static struct platform_device *ambarella_devices[] __initdata = {
 	&ambarella_uart1,
 	&ambarella_udc0,
 	&ambarella_wdt0,
+	&hero3black_leds,
 };
 
 /* ==========================================================================*/
-- 
1.7.9.5

